client/allocrunner/interfaces/runner_lifecycle.go

```diff
@@ -44,6 +44,7 @@ type RunnerUpdateHook interface {
 
 type RunnerUpdateRequest struct {
 	Alloc *structs.Allocation
+	Node  *structs.Node
```

Comment:
As above: I think this is static and therefore doesn't need to be included in Update calls.